2021-05-03 01:36:42 [scrapy.utils.log] INFO: Scrapy 2.4.1 started (bot: tutorial)
2021-05-03 01:36:42 [scrapy.utils.log] INFO: Versions: lxml 4.6.3.0, libxml2 2.9.10, cssselect 1.1.0, parsel 1.5.2, w3lib 1.21.0, Twisted 21.2.0, Python 3.8.8 (default, Feb 24 2021, 15:54:32) [MSC v.1928 64 bit (AMD64)], pyOpenSSL 20.0.1 (OpenSSL 1.1.1k  25 Mar 2021), cryptography 3.4.7, Platform Windows-10-10.0.19041-SP0
2021-05-03 01:36:42 [scrapy.utils.log] DEBUG: Using reactor: twisted.internet.selectreactor.SelectReactor
2021-05-03 01:36:42 [scrapy.crawler] INFO: Overridden settings:
{'BOT_NAME': 'tutorial',
 'LOG_FILE': 'logs\\default\\demo\\f25621ecab6c11ebb32904d4c4ede382.log',
 'NEWSPIDER_MODULE': 'tutorial.spiders',
 'SPIDER_MODULES': ['tutorial.spiders']}
2021-05-03 01:36:42 [scrapy.extensions.telnet] INFO: Telnet Password: 53483594f2600663
2021-05-03 01:36:42 [scrapy.middleware] INFO: Enabled extensions:
['scrapy.extensions.corestats.CoreStats',
 'scrapy.extensions.telnet.TelnetConsole',
 'scrapy.extensions.logstats.LogStats']
2021-05-03 01:36:42 [scrapy.middleware] INFO: Enabled downloader middlewares:
['scrapy.downloadermiddlewares.httpauth.HttpAuthMiddleware',
 'scrapy.downloadermiddlewares.downloadtimeout.DownloadTimeoutMiddleware',
 'scrapy.downloadermiddlewares.defaultheaders.DefaultHeadersMiddleware',
 'scrapy.downloadermiddlewares.useragent.UserAgentMiddleware',
 'scrapy.downloadermiddlewares.retry.RetryMiddleware',
 'scrapy.downloadermiddlewares.redirect.MetaRefreshMiddleware',
 'scrapy.downloadermiddlewares.httpcompression.HttpCompressionMiddleware',
 'scrapy.downloadermiddlewares.redirect.RedirectMiddleware',
 'scrapy.downloadermiddlewares.cookies.CookiesMiddleware',
 'scrapy.downloadermiddlewares.httpproxy.HttpProxyMiddleware',
 'scrapy.downloadermiddlewares.stats.DownloaderStats']
2021-05-03 01:36:42 [scrapy.middleware] INFO: Enabled spider middlewares:
['scrapy.spidermiddlewares.httperror.HttpErrorMiddleware',
 'scrapy.spidermiddlewares.offsite.OffsiteMiddleware',
 'scrapy.spidermiddlewares.referer.RefererMiddleware',
 'scrapy.spidermiddlewares.urllength.UrlLengthMiddleware',
 'scrapy.spidermiddlewares.depth.DepthMiddleware']
2021-05-03 01:36:42 [twisted] CRITICAL: Unhandled error in Deferred:
2021-05-03 01:36:42 [twisted] CRITICAL: 
Traceback (most recent call last):
  File "d:\anaconda\envs\pycharmprojects\lib\site-packages\twisted\internet\defer.py", line 1445, in _inlineCallbacks
    result = current_context.run(g.send, result)
  File "d:\anaconda\envs\pycharmprojects\lib\site-packages\scrapy\crawler.py", line 87, in crawl
    self.engine = self._create_engine()
  File "d:\anaconda\envs\pycharmprojects\lib\site-packages\scrapy\crawler.py", line 101, in _create_engine
    return ExecutionEngine(self, lambda _: self.stop())
  File "d:\anaconda\envs\pycharmprojects\lib\site-packages\scrapy\core\engine.py", line 70, in __init__
    self.scraper = Scraper(crawler)
  File "d:\anaconda\envs\pycharmprojects\lib\site-packages\scrapy\core\scraper.py", line 71, in __init__
    self.itemproc = itemproc_cls.from_crawler(crawler)
  File "d:\anaconda\envs\pycharmprojects\lib\site-packages\scrapy\middleware.py", line 53, in from_crawler
    return cls.from_settings(crawler.settings, crawler)
  File "d:\anaconda\envs\pycharmprojects\lib\site-packages\scrapy\middleware.py", line 35, in from_settings
    mw = create_instance(mwcls, settings, crawler)
  File "d:\anaconda\envs\pycharmprojects\lib\site-packages\scrapy\utils\misc.py", line 173, in create_instance
    instance = objcls(*args, **kwargs)
  File "D:\PycharmProjects\Movie_recommend\tutorial\tutorial\pipelines.py", line 14, in __init__
    self.conn = pymysql.connect(user='root', password='123456', database='how2java')
  File "d:\anaconda\envs\pycharmprojects\lib\site-packages\pymysql\connections.py", line 353, in __init__
    self.connect()
  File "d:\anaconda\envs\pycharmprojects\lib\site-packages\pymysql\connections.py", line 633, in connect
    self._request_authentication()
  File "d:\anaconda\envs\pycharmprojects\lib\site-packages\pymysql\connections.py", line 907, in _request_authentication
    auth_packet = self._read_packet()
  File "d:\anaconda\envs\pycharmprojects\lib\site-packages\pymysql\connections.py", line 725, in _read_packet
    packet.raise_for_error()
  File "d:\anaconda\envs\pycharmprojects\lib\site-packages\pymysql\protocol.py", line 221, in raise_for_error
    err.raise_mysql_exception(self._data)
  File "d:\anaconda\envs\pycharmprojects\lib\site-packages\pymysql\err.py", line 143, in raise_mysql_exception
    raise errorclass(errno, errval)
pymysql.err.OperationalError: (1049, "Unknown database 'how2java'")
